microservices-connector/cmd/router/main.go

```diff
@@ -477,6 +479,109 @@ func mcGraphHandler(w http.ResponseWriter, req *http.Request) {
+			err := r.ParseMultipartForm(64 << 20) // 64 MB is the default used by ParseMultipartForm
+			if err != nil {
+				http.Error(w, "Failed to parse multipart form", http.StatusBadRequest)
+				return
+			}
+			// Create a buffer to hold the new form data
+			var buf bytes.Buffer
+			writer := multipart.NewWriter(&buf)
+			// Copy all form fields from the original request to the new request
+			for key, values := range r.MultipartForm.Value {
+				for _, value := range values {
+					err := writer.WriteField(key, value)
+					if err != nil {
+						http.Error(w, "Failed to write form field", http.StatusInternalServerError)
+						return
+					}
+				}
+			}
+			// Copy all files from the original request to the new request
+			for key, fileHeaders := range r.MultipartForm.File {
+				for _, fileHeader := range fileHeaders {
+					file, err := fileHeader.Open()
+					if err != nil {
+						http.Error(w, "Failed to open file", http.StatusInternalServerError)
+						return
+					}
+					defer func() {
+						if err := file.Close(); err != nil {
+							log.Error(err, "error closing file")
+						}
+					}()
+					part, err := writer.CreateFormFile(key, fileHeader.Filename)
+					if err != nil {
+						http.Error(w, "Failed to create form file", http.StatusInternalServerError)
+						return
+					}
+					_, err = io.Copy(part, file)
+					if err != nil {
+						http.Error(w, "Failed to copy file", http.StatusInternalServerError)
+						return
+					}
+				}
+			}
+			err = writer.Close()
+			if err != nil {
+				http.Error(w, "Failed to close writer", http.StatusInternalServerError)
+				return
+			}
+			req, err := http.NewRequest(r.Method, serviceURL, &buf)
+			if err != nil {
+				http.Error(w, "Failed to create new request", http.StatusInternalServerError)
+				return
+			}
+			// Copy headers from the original request to the new request
+			for key, values := range r.Header {
+				for _, value := range values {
+					req.Header.Add(key, value)
+				}
+			}
+			req.Header.Set("Content-Type", writer.FormDataContentType())
+			client := &http.Client{}
+			resp, err := client.Do(req)
+			if err != nil {
+				http.Error(w, "Failed to send request to backend", http.StatusInternalServerError)
+				return
+			}
+			defer func() {
+				if err := resp.Body.Close(); err != nil {
+					log.Error(err, "error closing response body stream")
+				}
+			}()
+			// Copy the response headers from the backend service to the original client
+			for key, values := range resp.Header {
+				for _, value := range values {
+					w.Header().Add(key, value)
+				}
+			}
+			w.WriteHeader(resp.StatusCode)
+			// Copy the response body from the backend service to the original client
+			_, err = io.Copy(w, resp.Body)
+			if err != nil {
+				log.Error(err, "failed to copy response body")
+			}
+		}
+	}
+}
+
+func initializeRoutes() *http.ServeMux {
+	mux := http.NewServeMux()
+	mux.HandleFunc("/", mcGraphHandler)
+	mux.HandleFunc("/dataprep", mcDataHandler)
```

Comment:
maybe add a helper here? To indicate the correct path if user input unexpected ones?

Reply:
Hi @KfreeZ, I suggest that add a README for it, does it make sense?

Reply:
> maybe add a helper here? To indicate the correct path if user input unexpected ones?

@KfreeZ how would we do this? Need to keep all valid paths in some array and if user provides something not in valid list return a usage message?
Do we have a parent class from which our router paths are defined, perhaps we can add this valid paths functionality there. An enhancement feature?

Reply:
can we add a listService endpoint by default in router? in this way user can list all available endpoint throuth router. because even for dataprep, it will have 3 endpoints, like /dataprep /getDoc /deleteDoc etc.

Reply:
Hi @irisdingbj, as we discussed, I will raise an issue for dataprep to DCAI team, they should remove such verb. It's not the best practice for REST API design.

Reply:
Hi @irisdingbj, I opened an issue for dataprep REST API definition: https://github.com/opea-project/GenAIComps/issues/394